test/gtest/uct/ib/test_ib_pkey.cc

```diff
@@ -6,16 +6,36 @@
 #include <uct/ib/test_ib.h>
 
 
-class test_uct_ib_pkey : public test_uct_ib {
+class test_uct_ib_pkey : public test_uct_ib_with_specific_port {
 public:
     void init() {
+        test_uct_ib_with_specific_port::init();
+        /* Since we do `test_uct_ib::init-cleanup` in `send_recv_short()`,
+         * call `test_uct_ib::cleanup` to avoid entities overwritting */
+        test_uct_ib::cleanup();
+    }
+
+    void cleanup() {
+        /* Since we do `test_uct_ib::init-cleanup` in `send_recv_short()`,
+         * call `test_uct_ib::init` to restore entities for immediate
+         * cleanup inside `test_uct_ib_with_specific_port::cleanup`*/
         test_uct_ib::init();
+        test_uct_ib_with_specific_port::cleanup();
+    }
 
-        if (IBV_PORT_IS_LINK_LAYER_ETHERNET(&m_port_attr)) {
-            test_uct_ib::cleanup();
-            /* no pkeys for Ethernet */
-            UCS_TEST_SKIP_R("skip pkey test for port with Ethernet link type");
-        }
+    bool should_skip() {
+        /* no pkeys for Ethernet */
+        return IBV_PORT_IS_LINK_LAYER_ETHERNET(&m_port_attr);
+    }
+
+    void skip() {
+        UCS_TEST_SKIP_R("skip pkey test for port with Ethernet link type");
+    }
+
+    void send_recv_short() {
+        test_uct_ib::init();
```

Comment:
why is this needed here?

Reply:
to allow testing "invalid pkey" and "all available pkeys" cases:
1. "invalid pkey": we search for an invalid pkey value, modify config by the new pkey and test send-recv with it
2. "all available pkeys": go over all pkey table elements { takes pkeys with full membership, test send-recv with it }

Reply:
so maybe send_recv_short() in test_uct_ib should create the entity when it starts, and destroy the entity when it finishes? same for test_address_pack
we should avoid calling init()/cleanup() directly during test itself, they should be called only before and after the test respectively. any other behavior unexpected and will create confusion and inconsistency.